src/Tabs/Tabs.js

```diff
@@ -1,111 +1,129 @@
+                        if (!disabled && onClick) {
+                            onClick(e);
+                        }
+                    }}>
+                    {children}
+                </a>
+            );
+        } else if (children && React.isValidElement(children)) {
+            return React.cloneElement(children, {
+                ...tabLinkProps,
+                'aria-disabled': disabled,
+                className: className,
+                disabled: disabled,
+                onClick: (e) => {
+                    if (!disabled && onClick) {
+                        onClick(e);
+                    }
+                }
+            });
+        } else if (children) {
+            return children;
+        }
+    };
 
     return (
-        <ul
-            {...rest}
-            className={tabClasses}>
-            {children}
-        </ul>
+        <li {...rest}
+            className='fd-tabs__item'
+            key={id} >
+            {renderLink()}
+            {selectedId === id ? (
+                <p {...tabContentProps} className='fd-tabs__content'>{content}</p>
+            ) : null}
+        </li>
     );
 };
 
-Tabs.propTypes = {
+Tab.propTypes = {
+    id: PropTypes.string.isRequired,
     children: PropTypes.node,
-    className: PropTypes.string
+    className: PropTypes.string,
+    url: PropTypes.string
 };
 
-export class TabComponent extends Component {
+Tab.propDescriptions = {
+    children: 'Can be link text, an achor tag, or a react component like React Routers\'s `Link`.',
+    url: 'Creates an internal anchor when a child anchor is not provided.'
+};
+
+export class TabGroup extends Component {
     constructor(props) {
         super(props);
-        let initialStates = [];
 
-        initialStates = props.ids.forEach(ids => {
-            let obj = {};
-            let id = ids.id;
-            obj[id] = false;
-            return obj;
-        });
         this.state = {
-            selectedTab: '1',
-            tabStates: initialStates
+            selectedId: this.props.selectedId
         };
     }
 
     handleTabSelection = (e, id) => {
-        let iStates = Object.assign({}, this.state.tabStates);
-        iStates[id.id] = !iStates[id.id];
-        this.setState({ tabStates: iStates });
-        this.setState({ selectedTab: id.id });
+        this.setState({ selectedId: id });
     };
 
-    getLinkClasses = (id) => {
+    getLinkClasses = (id, className) => {
         return classnames(
+            className,
             'fd-tabs__link',
             {
-                'is-selected': this.state.selectedTab === id
+                'is-selected': this.state.selectedId === id
             }
         );
     }
 
+    renderTabs = (children) => {
+        return React.Children.map(children, (child) => {
+            const classes = this.getLinkClasses(child.props.id, child.props.className);
+
+            return React.cloneElement(child, {
+                selectedId: this.state.selectedId,
```

Comment:
Instead of passing the `selectedId` to each child `Tab`, what if the comparison of `child.props.id` to `selectedId` occurred here and the `Tab` component was simply passed a `selected` prop?  Keep more of the decision making in this component (that is carrying state) and leave the `Tab` as a dumb component.